Formula/cfitsio.rb

```diff
@@ -39,4 +40,18 @@ def install
       end
     end
   end
+
+  test do
+    cp Dir["#{pkgshare}/testprog/testprog*"], testpath
+    flags = (ENV.cflags || "").split + (ENV.cppflags || "").split + (ENV.ldflags || "").split
```

Comment:
Okay, I think I see what it's doing. It's not picking up user-defined variables. The tests run under StdEnv, which first sanitizes these vars and [may then set up new values for them](https://github.com/Homebrew/brew/blob/4b2c4ef25835a87b5bac4fb26eae91dbfed93863/Library/Homebrew/extend/ENV/std.rb#L31-L47).

So, this line pulls the StdEnv-supplied values out of the env vars and puts them in the `cc` command line. That sounds like a way of adding support for compilers which don't respect those environment variables. Are there any? Or maybe it's doing it so all those flags show up in the test execution output?